tests/integration/Mvc/Model/UnderscoreSetCest.php

```diff
@@ -12,20 +12,301 @@
+        $model->id = 123;
+
+        $I->assertEquals(
+            123,
+            $model->getId()
+        );
+
+        $associativeArray = [
+            'firstName' => 'First name',
+            'lastName'  => 'Last name'
+        ];
+
+        $model->name = $associativeArray;
+
+        $I->assertEquals(
+            $associativeArray,
+            $model->getName()
+        );
+
+        $model->text = 'MyText';
+
+        $I->assertEquals(
+            'MyText',
+            $model->getText()
+        );
+
+        $I->assertEquals(
+            [
+                'sel_id'   => 123,
+                'sel_name' => $associativeArray,
+                'sel_text' => 'MyText'
+            ],
+            $model->toArray()
+        );
+    }
+
+    /**
+     * Tests Phalcon\Mvc\Model :: __set() with belongs-to related record
+     *
+     * @param IntegrationTester $I
+     *
+     * @author Balázs Németh <https://github.com/zsilbi>
+     * @since  2019-05-02
+     */
+    public function mvcModelUnderscoreSetWithBelongsToRelatedRecord(IntegrationTester $I)
+    {
+        $I->wantToTest('Tests Mvc\Model - __set() with belongs-to related record');
+
+        $robotPart = new Models\RobotsParts();
+        $robotPart->robot = new Models\Robots();
+
+        $robot = $robotPart->robot;
+
+        $I->assertInstanceOf(Models\Robots::class, $robot);
+        $I->assertEquals($robotPart->getDirtyState(), Model::DIRTY_STATE_TRANSIENT);
+    }
+
+    /**
+     * Tests Phalcon\Mvc\Model :: __set() with an array as properties of a belongs-to related record
+     *
+     * @param IntegrationTester $I
+     *
+     * @author Balázs Németh <https://github.com/zsilbi>
+     * @since  2019-05-02
+     */
+    public function mvcModelUnderscoreSetWithArrayOfBelongsToRelatedRecord(IntegrationTester $I)
+    {
+        $I->wantToTest('Tests Mvc\Model - __set() with an array as properties of a belongs-to related record');
+
+        $robotPart = new Models\RobotsParts();
+        $robotPart->robot = ['name' => 'TestRobotName'];
+
+        $robot = $robotPart->robot;
+
+        $I->assertInstanceOf(Models\Robots::class, $robot);
+        $I->assertEquals($robotPart->getDirtyState(), Model::DIRTY_STATE_TRANSIENT);
+        $I->assertEquals($robot->name, 'TestRobotName');
+    }
+
+    /**
+     * Tests Phalcon\Mvc\Model :: __set() with has-one related record
+     *
+     * @param IntegrationTester $I
+     *
+     * @author Balázs Németh <https://github.com/zsilbi>
+     * @since  2019-05-02
+     */
+    public function mvcModelUnderscoreSetWithHasOneRelatedRecord(IntegrationTester $I)
+    {
+        $I->wantToTest('Tests Mvc\Model - __set() with has-one related record');
+
+        $customer = new Models\Customers();
+        $customer->user = new Models\Users();
+
+        $user = $customer->user;
+
+        $I->assertInstanceOf(Models\Users::class, $user);
+        $I->assertEquals($customer->getDirtyState(), Model::DIRTY_STATE_TRANSIENT);
```

Comment:
Assertions are `($expected, $actual)` so it should be:

```php
$I->assertEquals(
    Model::DIRTY_STATE_TRANSIENT,
    $customer->getDirtyState()
);
```

And space out the assertions on multiple lines to make it easier to read. :wink:

Reply:
Thank you, @SidRoberts :)